FactorioBlueprintStringRenderer/src/com/demod/fbsr/LogisticGridCell.java

```diff
@@ -58,6 +58,10 @@ public void addOutput(String itemName) {
 		if (!outputs.isPresent()) {
 			outputs = Optional.of(new LinkedHashSet<>());
 		}
+		if (itemName.equals("UNKNOWN")) {
+			System.out.println("WARNING: outputs contains UNKNOWN");
+			return;
+		}

```

Comment:
BSFilter contains the line `name = BSUtils.optString(json, "name").orElse("UNKNOWN")`. That happens when a filter is on quality but not entity, like a filter for quality >= uncommon. That UNKNOWN string makes its way here, and eventually there is warn logging about the unknown entity.

Reply:
It seems like it might be better to leave these warnings out for now, especially since perfectly valid blueprints that use filters with only quality quantified can go down this code path.  Quality is still pending on the todo list, so it will be addressed when that task gets prioritized.

Reply:
Changed this line to return without logging.